druid-shell/src/platform/x11/keycodes.rs

```diff
@@ -0,0 +1,335 @@
+            73 => KeyCode::F7,
+            74 => KeyCode::F8,
+            75 => KeyCode::F9,
+            76 => KeyCode::F10,
+            95 => KeyCode::F11,
+            96 => KeyCode::F12,
+
+            107 => KeyCode::PrintScreen,
+            118 => KeyCode::Insert,
+            119 => KeyCode::Delete,
+            49 => KeyCode::Quote,
+
+            10 => KeyCode::Numpad1,
+            11 => KeyCode::Numpad2,
+            12 => KeyCode::Numpad3,
+            13 => KeyCode::Numpad4,
+            14 => KeyCode::Numpad5,
+            15 => KeyCode::Numpad6,
+            16 => KeyCode::Numpad7,
+            17 => KeyCode::Numpad8,
+            18 => KeyCode::Numpad9,
+            19 => KeyCode::Numpad0,
+
+            20 => KeyCode::LeftBracket,
+            21 => KeyCode::RightBracket,
+            22 => KeyCode::Backspace,
+
+            23 => KeyCode::Tab,
+
+            24 => KeyCode::KeyQ,
+            25 => KeyCode::KeyW,
+            26 => KeyCode::KeyE,
+            27 => KeyCode::KeyR,
+            28 => KeyCode::KeyT,
+            29 => KeyCode::KeyY,
+            30 => KeyCode::KeyU,
+            31 => KeyCode::KeyI,
+            32 => KeyCode::KeyO,
+            33 => KeyCode::KeyP,
+
+            34 => KeyCode::Slash,
+            35 => KeyCode::Equals,
+            36 => KeyCode::Return,
+
+            66 => KeyCode::CapsLock,
+
+            38 => KeyCode::KeyA,
+            39 => KeyCode::KeyS,
+            40 => KeyCode::KeyD,
+            41 => KeyCode::KeyF,
+            42 => KeyCode::KeyG,
+            43 => KeyCode::KeyH,
+            44 => KeyCode::KeyJ,
+            45 => KeyCode::KeyK,
+            46 => KeyCode::KeyL,
+
+            47 => KeyCode::Semicolon,
+            48 => KeyCode::Comma,
+            // 49 => KeyCode::Hash,
+            50 => KeyCode::LeftShift,
+            94 => KeyCode::Backslash,
+
+            52 => KeyCode::KeyZ,
+            53 => KeyCode::KeyX,
+            54 => KeyCode::KeyC,
+            55 => KeyCode::KeyV,
+            56 => KeyCode::KeyB,
+            57 => KeyCode::KeyN,
+            58 => KeyCode::KeyM,
+
+            59 => KeyCode::Comma,
+            60 => KeyCode::Period,
+            61 => KeyCode::NumpadMultiply,
+            64 => KeyCode::RightShift,
+
+            37 => KeyCode::LeftControl,
+            133 => KeyCode::LeftMeta,
+            // TODO(x11/keycodes): case 64 is duplicated; which one is correct?
+            //     What should the other be?
+            // 64 => KeyCode::LeftAlt,
+            65 => KeyCode::Space,
+            108 => KeyCode::RightAlt,
+            105 => KeyCode::RightControl,
+
+            111 => KeyCode::ArrowUp,
+            113 => KeyCode::ArrowLeft,
+            114 => KeyCode::ArrowRight,
+            116 => KeyCode::ArrowDown,
+
+            _ => {
+                log::warn!("Warning: unknown keyval {}", raw);
+                KeyCode::Unknown(raw)
+            }
+        }
+    }
+}
+
+impl Into<StrOrChar> for KeyCode {
```

Comment:
a few things going on here: `StrOrChar` is not intended to be used directly; it is a way to allow the `KeyEvent` constructor to take either a `str` or a `char`.

More importantly, `KeyCode` cannot be used on its own to determine the string value of a key press. The `KeyCode` refers to a specific physical key; to interpret this we need to map it through the user's current locale and keyboard layout. (I think we talked about this? It's on zulip somewhere)

Reply:
(I'm unsure what to do here too -- paging @perlindgren for help)